internal/config/config.go

```diff
@@ -141,8 +141,9 @@ type Slack struct {
 
 // Artifacts represents the test artifacts configuration.
 type Artifacts struct {
-	Download ArtifactDownload `yaml:"download,omitempty" json:"download"`
-	Cleanup  bool             `yaml:"cleanup,omitempty" json:"cleanup"`
+	Retain   map[string]string `yaml:"retain,omitempty" json:"retain"`
```

Comment:
I think this needs some validation. Re. naming at least. That is needs to be a zip file etc..
We don't want to waste VMs for illegal input.